depthmapXcli/runmethods.cpp

```diff
@@ -257,6 +257,63 @@ namespace dm_runmethods
 
     }
 
+    void runSegmentAnalysis(const CommandLineParser &clp, const SegmentParser &sp, IPerformanceSink &perfWriter)
+    {
+        auto mGraph = loadGraph(clp.getFileName().c_str(), perfWriter);
+
+        auto state = mGraph->getState();
+
+        std::cout << "Running segment analysis... " << std::flush;
+        Options options;
+        options.radius_list = genshim::toPVector(sp.getRadii());
+        options.choice = sp.includeChoice();
+        options.tulip_bins = sp.getTulipBins();
+        switch(sp.getRadiusType()) {
+            case SegmentParser::RadiusType::SEGMENT_STEPS: {
+                options.radius_type = Options::RADIUS_STEPS;
+                break;
+            }
+            case SegmentParser::RadiusType::METRIC: {
+                options.radius_type = Options::RADIUS_METRIC;
+                break;
+            }
+            case SegmentParser::RadiusType::ANGULAR: {
+                options.radius_type = Options::RADIUS_ANGULAR;
+                break;
+            }
+            case SegmentParser::RadiusType::NONE:
+                break;
+        }
+        switch(sp.getAnalysisType()) {
+            case SegmentParser::AnalysisType::ANGULAR_TULIP: {
+                DO_TIMED("Segment tulip analysis", mGraph->analyseSegments(0, options))
+                break;
+            }
+            case SegmentParser::AnalysisType::ANGULAR_FULL: {
+                DO_TIMED("Segment angular analysis", mGraph->analyseSegments(0, options))
+                break;
+            }
+            case SegmentParser::AnalysisType::TOPOLOGICAL: {
+                options.output_type = 0; // derived from the GUI
+                DO_TIMED("Segment topological", mGraph->analyseTopoMet(0, options))
+                break;
+            }
+            case SegmentParser::AnalysisType::METRIC: {
+                options.output_type = 1; // derived from the GUI
+                DO_TIMED("Segment metric", mGraph->analyseTopoMet(0, options))
+                break;
+            }
+            case SegmentParser::AnalysisType::NONE:
```

Comment:
Does this make sense to silently not do anything if the type is NONE and then pretty much write out a copy of the graph as result?

Reply:
This is handled in segmentparser.cpp (127):
    if (getAnalysisType() == AnalysisType::NONE)
    {
        throw CommandLineException("No analysis type given");
    }